sysvabi64/sysvabi64.rst

```diff
@@ -749,12 +749,27 @@ open-source compilers.
 Medium code model
 -----------------
 
-Symbols smaller than 64KiB (value TBD) are treated like the small code model.
-Symbols which are larger than 64KiB, are ``common`` or of unknown size use
-sections ``.ldata``, ``.lbss`` and ``.lrodata``. These sections are placed after
-all other sections and can be unlimited in size. Symbols in them are addressed
-via the GOT. Linkers may relax GOT indirections into PC-relative addressing for
-non pre-emptible symbols within 4GiB.
+
+The medium code model is suitable for programs that do not have large amounts
+of executable code, but do contain large static data objects. The medium code
+model separates data into small data and large data. Small data is addressed
+in the same way as the small code model, with the same maximum size
+restrictions as the small code model. Large data is addressed via the GOT and
+does not have a maximum size limit.
+
+Data is defined as large data if any of the following are true:
+
+- The data is >= 64 KiB in size.
+- The data is of unknown size.
+- The data has type ``SHN_COMMON``.
+
+All other data is small data.
+
+Large data is placed in sections with a ``l`` prefix, for example ``.lbss``,
```

Comment:
I've made a few suggestions making the distinction between sections in objects and at link time. I've included a suggestion for ``.data.rel.ro`` (thanks MaskRay) I had forgotten that ``ld.so`` etc. only support one RELRO segment.  

The object producer places large data in sections with an additional ``l`` prefix, for example ``.lbss``, ``.ldata``, and ``.lrodata``. Both small and large RELRO data are placed in ``.data.rel.ro`` as many loaders only support one ``PT_GNU_RELRO`` program header.

At static link time the large data sections must be placed after all small data sections. The static linker allocates storage for SHN_COMMON symbols at the end of the ``.bss`` section if it is the last small data section, otherwise it is allocated in ``.lbss``.

Reply:
It's a shame there can only be one RELRO segment, however limiting .data.rel.ro to 4GB doesn't seem like a big restriction. The distinction between object producers/consumers looks good indeed.